src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1141,6 +1147,14 @@ public boolean hasAttribute(final String tag) {
         return getAttribute(tag)!=null;
     }
 
+    /**
+     * @return {@code true} if the SAM record has the requested attribute set, {@code false} otherwise.
+     */
+    public boolean hasAttribute(final SAMTag tag) {
+        return getAttribute(tag.name())!=null;
```

Comment:
these new methods should use `getAttribute(tag.getBinaryTag())` when possible to avoid redundant conversions

Reply:
👍